### PR TITLE
CTCP-1154 create office of exit CYA

### PR DESCRIPTION
**Pull request by kenneth-rayner**

### CODE REVIEW

File: app/services/CountriesService.scala [hunk 1]
```diff
@@ -76,10 +77,12 @@ class CountriesService @Inject() (referenceDataConnector: ReferenceDataConnector
       .getCustomsSecurityAgreementAreaCountries()
       .map(sort)
 
-  def getCountryCodesCTC()(implicit hc: HeaderCarrier): Future[CountryList] =
+  def getCountryCodesCTC()(implicit hc: HeaderCarrier): Future[CountryList] = {
+    println("I was here")
```
Comment (kevincoatsworth) [inline app/services/CountriesService.scala:81]: You've accidentally left this in

File: app/controllers/routeDetails/routing/AddAnotherCountryOfRoutingController.scala [hunk 2]
```diff
@@ -50,7 +50,7 @@ class AddAnotherCountryOfRoutingController @Inject() (
     implicit request =>
       val (countries, numberOfCountries, allowMoreCountries) = viewData
       numberOfCountries match {
-        case 0 => Redirect(routes.BindingItineraryController.onPageLoad(lrn, NormalMode))
+        case 0 => Redirect(routes.AddCountryOfRoutingYesNoController.onPageLoad(lrn, NormalMode))
```
Comment (jackheslop96) [inline app/controllers/routeDetails/routing/AddAnotherCountryOfRoutingController.scala:53]: I believe this change is incorrect. There is a scenario in which the user can start adding countries of routing without ever having to answer the "add country of routing yes/no" question. Redirecting them to here would go against our navigation logic. Sending them to binding itinerary ensures that whatever they then subsequently answer for that question will take them to the correct place

File: app/controllers/routeDetails/transit/index/OfficeOfTransitController.scala [hunk 3]
```diff
@@ -56,7 +56,7 @@ class OfficeOfTransitController @Inject() (
           val country = request.arg
           service.getCustomsOfficesForCountry(country.code).map {
             customsOfficeList =>
-              val form = formProvider("routeDetails.transit.officeOfTransit", customsOfficeList, country.description)
+              val form = formProvider("routeDetails.transit.officeOfExit", customsOfficeList, country.description)
```
Comment (jackheslop96) [inline app/controllers/routeDetails/transit/index/OfficeOfTransitController.scala:59]: Was this change intended?

File: app/services/CountriesService.scala (hunk 1 above)
Comment (jackheslop96) [inline app/services/CountriesService.scala:81]: Still needed?

File: app/utils/cyaHelpers/routeDetails/TransitCheckYourAnswersHelper.scala [hunk 4]
```diff
@@ -41,11 +41,14 @@ class TransitCheckYourAnswersHelper(
     buildListItems(OfficeOfTransitCountriesSection) {
       position =>
         val index = Index(position)
-        buildListItem[OfficeOfTransitDomain, Country](
-          page = OfficeOfTransitCountryPage(index),
+        buildListItem[OfficeOfTransitDomain, CustomsOffice](
+          page = OfficeOfTransitPage(index),
```
Comment (jackheslop96) [inline app/utils/cyaHelpers/routeDetails/TransitCheckYourAnswersHelper.scala:45]: This might cause issues. Could you do me a favour and run this locally to see what happens if you start adding an office of transit but only enter the country? Then click back to the add-another page

File: app/views/routeDetails/officeOfExit/CheckOfficeOfExitAnswersView.scala.html [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+    @heading(
+        heading =messages("routeDetails.officeOfExit.checkOfficeOfExitAnswers.heading"),
+        caption = Some(messages("routeDetails.caption"))
+    )
+
+    @answerSections(sections)
+
+    @formHelper(action = routes.CheckOfficeOfExitAnswersController.onSubmit(lrn, index), 'autoComplete -> "off") {
+    @button(attributes = Map("id" -> "submit"))
```
Comment (jackheslop96) [inline app/views/routeDetails/officeOfExit/CheckOfficeOfExitAnswersView.scala.html:46]: Indentation (sorry 😄 )

File: app/utils/cyaHelpers/routeDetails/TransitCheckYourAnswersHelper.scala (hunk 4 above)
Comment (kenneth-rayner) [reply app/utils/cyaHelpers/routeDetails/TransitCheckYourAnswersHelper.scala:45]: This was a minor bug we noticed and decided to to fix in this branch. The add office of transit page showed the countries rather than the offices in the summary. In the UI it seems to be ok when hitting the back button

Comment (jackheslop96) [reply app/utils/cyaHelpers/routeDetails/TransitCheckYourAnswersHelper.scala:45]: You should notice that when adding an office of transit with this approach, if you only enter a country and click back before selecting an office, it will NOT appear on the add-another page. Essentially we're introducing 'hidden' in progress offices of transit which will cause a whole host of issues. The 'page' that we pass in should always be the first page in the journey.

A change got merged in as part of CTCP-960 to use .label as the formatting method. This ensures that the customs office is rendered on the add-another page for completed offices

File: app/controllers/routeDetails/transit/index/OfficeOfTransitController.scala [hunk 6]
```diff
@@ -56,7 +56,7 @@ class OfficeOfTransitController @Inject() (
           val country = request.arg
           service.getCustomsOfficesForCountry(country.code).map {
             customsOfficeList =>
-              val form = formProvider("routeDetails.transit.officeOfTransit", customsOfficeList, country.description)
+              val form = formProvider("routeDetails.transit.officeOTransit", customsOfficeList, country.description)
```
Comment (jackheslop96) [inline app/controllers/routeDetails/transit/index/OfficeOfTransitController.scala:59]: Looks to be an unintentional diff

File: app/views/routeDetails/officeOfExit/CheckOfficeOfExitAnswersView.scala.html [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    )
+
+    <h2 class="govuk-heading-m">@messages("routeDetails.officeOfExit.checkOfficeOfExitAnswers.subHeading")</h2>
+
+    @answerSections(sections)
+
+
+    @formHelper(action = routes.CheckOfficeOfExitAnswersController.onSubmit(lrn, index), 'autoComplete -> "off") {
+        @button(attributes = Map("id" -> "submit"))
+        }
```
Comment (jackheslop96) [inline app/views/routeDetails/officeOfExit/CheckOfficeOfExitAnswersView.scala.html:50]: Closing bracket needs un-indenting (sorry again! 😅 )

**Review by jackheslop96 — APPROVED** [review]
